### PR TITLE
Add localization methods for weekday and month

### PR DESCRIPTION
**Pull request by borkweb**
This work is to aim to avoid the need for us to localize months and weekdays.

Sort of related to: https://central.tri.be/issues/70068

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | empty() and isset() must only contain variables (PHP 5.2 compatibility) | [Date_Utils.php:693](https://github.com/moderntribe/tribe-common/blob/928a63db382f9f94c9edf52c2ec5056a4bdb599a/src/Tribe/Date_Utils.php#L693)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Date_Utils.php [hunk 1]
```diff
@@ -666,6 +666,154 @@ private static function build_localized_weekdays() {
...
+
+				// if the month integer falls out of range, bail without localizing
+				if ( 0 > $month_int || 12 < $month_int ) {
+					return $month;
+				}
+
+				$month_formatted = str_pad( $month_formatted, 2, '0', STR_PAD_LEFT );
+
+				// get the appropriately formatted month name (we don't care about the year, let's use 2000)
+				$month_formatted = date( $date_format_for_month, strtotime( '2000-' . $month_formatted . '-10' ) );
```
Comment (barryhughes) [inline src/Tribe/Date_Utils.php]: Could this be problematic depending on the locale settings for PHP itself (ie, if `set_locale()` had been called earlier/if modified via the relevant .ini file)?

Comment (barryhughes) [inline src/Tribe/Date_Utils.php]: ...That is, could PHP's locale be at odds with WordPress's and so we'd have a mismatch?

Comment (borkweb) [inline src/Tribe/Date_Utils.php]: I suppose you are right. I guess I should go the laborious route of mapping months using arrays like I did with weekdays.

Comment (barryhughes) [inline src/Tribe/Date_Utils.php]: For months we're only concerned with two things - the full name and the abbreviation. Seems like we can get those with less work by doing:

```php
// Full
$wp_locale->get_month( 12 ); 

// Abbrev
$wp_locale->get_month_abbrev( 
    $wp_locale->get_month( 12 )
);
```

The problem there is efficiency, but just like you're doing we could cache the relevant data within a static field. We'd also eliminate this problem.

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | empty() and isset() must only contain variables (PHP 5.2 compatibility) | [Date_Utils.php:569](https://github.com/moderntribe/tribe-common/blob/b269c474651f6c6e609802ef9c812f64390249bd/src/Tribe/Date_Utils.php#L569)
:x: | empty() and isset() must only contain variables (PHP 5.2 compatibility) | [Date_Utils.php:601](https://github.com/moderntribe/tribe-common/blob/b269c474651f6c6e609802ef9c812f64390249bd/src/Tribe/Date_Utils.php#L601)
:x: | empty() and isset() must only contain variables (PHP 5.2 compatibility) | [Date_Utils.php:718](https://github.com/moderntribe/tribe-common/blob/b269c474651f6c6e609802ef9c812f64390249bd/src/Tribe/Date_Utils.php#L718)
:x: | empty() and isset() must only contain variables (PHP 5.2 compatibility) | [Date_Utils.php:789](https://github.com/moderntribe/tribe-common/blob/b269c474651f6c6e609802ef9c812f64390249bd/src/Tribe/Date_Utils.php#L789)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Date_Utils.php (hunk 1 above)
Comment (borkweb) [inline src/Tribe/Date_Utils.php]: I've reworked this (and simplified the weekday logic based on my reworking), though I've opted not to make direct calls to `$wp_locale` from within these new `wp_locale_x()` methods.

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | empty() and isset() must only contain variables (PHP 5.2 compatibility) | [Date_Utils.php:569](https://github.com/moderntribe/tribe-common/blob/ed682445ef95a336fad2ff25f27339ab974575a6/src/Tribe/Date_Utils.php#L569)
:x: | empty() and isset() must only contain variables (PHP 5.2 compatibility) | [Date_Utils.php:601](https://github.com/moderntribe/tribe-common/blob/ed682445ef95a336fad2ff25f27339ab974575a6/src/Tribe/Date_Utils.php#L601)
:x: | empty() and isset() must only contain variables (PHP 5.2 compatibility) | [Date_Utils.php:718](https://github.com/moderntribe/tribe-common/blob/ed682445ef95a336fad2ff25f27339ab974575a6/src/Tribe/Date_Utils.php#L718)
:x: | empty() and isset() must only contain variables (PHP 5.2 compatibility) | [Date_Utils.php:789](https://github.com/moderntribe/tribe-common/blob/ed682445ef95a336fad2ff25f27339ab974575a6/src/Tribe/Date_Utils.php#L789)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | empty() and isset() must only contain variables (PHP 5.2 compatibility) | [Date_Utils.php:569](https://github.com/moderntribe/tribe-common/blob/cc0badcf5af338e3ad8c09bc76494377bd31c689/src/Tribe/Date_Utils.php#L569)
:x: | empty() and isset() must only contain variables (PHP 5.2 compatibility) | [Date_Utils.php:601](https://github.com/moderntribe/tribe-common/blob/cc0badcf5af338e3ad8c09bc76494377bd31c689/src/Tribe/Date_Utils.php#L601)
:x: | empty() and isset() must only contain variables (PHP 5.2 compatibility) | [Date_Utils.php:722](https://github.com/moderntribe/tribe-common/blob/cc0badcf5af338e3ad8c09bc76494377bd31c689/src/Tribe/Date_Utils.php#L722)
:x: | empty() and isset() must only contain variables (PHP 5.2 compatibility) | [Date_Utils.php:795](https://github.com/moderntribe/tribe-common/blob/cc0badcf5af338e3ad8c09bc76494377bd31c689/src/Tribe/Date_Utils.php#L795)

> via jenkins/codesniffer-phpcs

**Comment by barryhughes** [comment]
🍶

**Pull request merged**